eclass/dotnet-utils.eclass

```diff
@@ -0,0 +1,216 @@
+	7|8) ;;
+	*) die "${ECLASS}: EAPI ${EAPI} unsupported." ;;
+esac
+
+if [[ ! ${_DOTNET_UTILS_ECLASS} ]]; then
+_DOTNET_UTILS_ECLASS=1
+
+inherit edo multiprocessing
+
+# @ECLASS_VARIABLE: DOTNET_SLOT
+# @REQUIRED
+# @PRE_INHERIT
+# @DESCRIPTION:
+# Allows to choose a slot for dotnet
+
+if [[ ! ${DOTNET_SLOT} ]]; then
+	die "${ECLASS}: DOTNET_SLOT not set"
+fi
+
+# Temporary, use the virtual once you can have multiple virtuals installed at once
+BDEPEND+=" || ( dev-dotnet/dotnet-sdk-bin:${DOTNET_SLOT} dev-dotnet/dotnet-sdk:${DOTNET_SLOT} )"
+
+IUSE+=" debug"
+
+# @ECLASS_VARIABLE: DOTNET_EXECUTABLE
+# @INTERNAL
+# @DESCRIPTION:
+# Sets to the valid path of a dotnet executable
+
+# @ECLASS_VARIABLE: DOTNET_OUTPUT
+# @OUTPUT_VARIABLE
+# @DESCRIPTION:
+# Path of the output directory, if unset, is set during src_compile to:
+# "${WORKDIR}"/${P}_net${DOTNET_SLOT}_${DOTNET_CONFIGURATION}/${PN}
+
+# @ECLASS_VARIABLE: DOTNET_CONFIGURATION
+# @DEFAULT_UNSET
+# @DESCRIPTION:
+# Configuration value passed to dotnet in the compile phase.
+# Defaults to Debug or Release, depending on the debug useflag.
+
+# @ECLASS_VARIABLE: DOTNET_CLI_TELEMETRY_OPTOUT
+# @OUTPUT_VARIABLE
+# @DESCRIPTION:
+# Disables telemetry on dotnet.
+export DOTNET_CLI_TELEMETRY_OPTOUT=1
+
+# @ECLASS_VARIABLE: MSBUILDDISABLENODEREUSE
+# @OUTPUT_VARIABLE
+# @DESCRIPTION:
+# Stops the dotnet node after the build is done.
+export MSBUILDDISABLENODEREUSE=1
+
+# @ECLASS_VARIABLE: DOTNET_NOLOGO
+# @OUTPUT_VARIABLE
+# @DESCRIPTION:
+# Disables the header logo when running dotnet commands.
+export DOTNET_NOLOGO=1
+
+# Needed otherwise the binaries may break
+RESTRICT+=" strip"
+
+# @ECLASS_VARIABLE: NUGETS
+# @DEFAULT_UNSET
+# @DESCRIPTION:
+# String containing all nuget packages that need to be downloaded.
+# Used by nuget_uris()
+#
+# Example:
+# @CODE
+# NUGETS="
+# 	ImGui.NET-1.87.2
+# 	Config.Net-4.19.0
+# "
+#
+# inherit dotnet-utils
+#
+# ...
+#
+# SRC_URI="$(nuget_uris)"
+# @CODE
+
+# @FUNCTION: nuget_uris
+# @USAGE: <nuget...>
+# @DESCRIPTION:
+# Generates the URIs to put in SRC_URI to help fetch dependencies.
+# If no arguments provided, uses NUGETS variable.
+nuget_uris() {
+	local -r regex='^([a-zA-Z0-9_.-]+)-([0-9]+\.[0-9]+\.[0-9]+.*)$'
+	local nuget nugets
+
+	if (( $# != 0 )); then
+		nugets="${@}"
+	elif [[ ${NUGETS} ]]; then
+		nugets="${NUGETS}"
+	else
+		eerror "NUGETS variable is not defined and nothing passed as argument"
+		die "${FUNCNAME}: Can't generate SRC_URI from empty input"
+	fi
+
+	for nuget in ${nugets}; do
+		local name version url
+		[[ ${nuget} =~ ${regex} ]] || die "Could not parse name and version from nuget: $nuget"
+		name="${BASH_REMATCH[1]}"
+		version="${BASH_REMATCH[2]}"
+		url="https://api.nuget.org/v3-flatcontainer/${name}/${version}/${name}.${version}.nupkg"
+		echo "${url}"
+	done
+}
+
+# @FUNCTION: dotnet-utils_pkg_setup
+# @DESCRIPTION:
+# Sets up DOTNET_EXECUTABLE variables for later use in edotnet.
+dotnet-utils_pkg_setup() {
+	if use amd64; then
```

Comment:
afaik the re is also ...-musl runtime

Reply:
https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.NETCore.Platforms/src/runtime.json

Turns out there is a lot of runtimes, but for linux, mainly amd64, x86, arm, arm64 and ppc64le, and all have -musl variants. (I didn't add the ppc64le because I'm not sure if gentoo's ppc supports le)

Reply:
We support ppc64 in both big endian and little endian, but in Gentoo, ppc (32-bit) is only big endian